### PR TITLE
docs: Add all available HF/sentence transformers embedding models list

### PR DESCRIPTION
**Pull request by AyushExel**
Solves -  https://github.com/lancedb/lancedb/issues/968

### CODE REVIEW

**Comment by sankalp1999** [comment]
some formatting issues are there

**Comment by sankalp1999** [comment]
can possibly change "Some available models ..." to "Check out available sentence-transformer models here!"

File: docs/src/embeddings/default_embedding_functions.md [hunk 1]
```diff
@@ -19,6 +19,139 @@ Allows you to set parameters when registering a `sentence-transformers` object.
...
+    - sentence-transformers/all-MiniLM-L12-v1
+    ```
+
+!!! info
+    You can also load many other model architectures from the library. For example models from sources such as BAAI, nomic, salesforce research, etc.
+    See this HF hub page for all [supported models](https://huggingface.co/models?library=sentence-transformers).
+
+Visit sentence-transformers [HuggingFace HUB](https://huggingface.co/sentence-transformers) page for more information on the available models.
 ```python
 db = lancedb.connect("/tmp/db")
```
Comment (sankalp1999) [inline docs/src/embeddings/default_embedding_functions.md:156]: can demonstrate some integration in this example itself

Comment (AyushExel) [reply docs/src/embeddings/default_embedding_functions.md]: ok done

**Comment by sankalp1999** [comment]
LGTM

File: docs/src/embeddings/default_embedding_functions.md [hunk 2]
```diff
@@ -151,28 +151,31 @@ Allows you to set parameters when registering a `sentence-transformers` object.
...
-Visit sentence-transformers [HuggingFace HUB](https://huggingface.co/sentence-transformers) page for more information on the available models.
-```python
-db = lancedb.connect("/tmp/db")
-registry = EmbeddingFunctionRegistry.get_instance()
-func = registry.get("sentence-transformers").create(device="cpu")
-
-class Words(LanceModel):
-    text: str = func.SourceField()
-    vector: Vector(func.ndims()) = func.VectorField()
+!!! note "BAAI Embeddings example"
```
Comment (sankalp1999) [inline docs/src/embeddings/default_embedding_functions.md:154]: people know the term bge more than BAAI.
can possibly change to BAAI bge-small-en-v1.5  change

File: docs/src/embeddings/default_embedding_functions.md [hunk 3]
```diff
@@ -151,28 +151,31 @@ Allows you to set parameters when registering a `sentence-transformers` object.
...
+    Here is an example that uses BAAI embedding model from the HuggingFace Hub [supported models](https://huggingface.co/models?library=sentence-transformers)
+    ```python
+    db = lancedb.connect("/tmp/db")
+    registry = EmbeddingFunctionRegistry.get_instance()
+    func = registry.get("sentence-transformers").create(name="BAAI/bge-small-en-v1.5", device="cpu")
+
```
Comment (sankalp1999) [inline docs/src/embeddings/default_embedding_functions.md:160]: can use the term `model`. i know it's embeddingfunction but it's still more intuitive than `func`

File: docs/src/embeddings/default_embedding_functions.md [hunk 4]
```diff
@@ -19,27 +19,163 @@ Allows you to set parameters when registering a `sentence-transformers` object.
...
+    - sentence-transformers/clip-ViT-B-16
+    - sentence-transformers/use-cmlm-multilingual
+    - sentence-transformers/all-MiniLM-L12-v1
+    ```
 
-class Words(LanceModel):
-    text: str = func.SourceField()
-    vector: Vector(func.ndims()) = func.VectorField()
+!!! info
+    You can also load many other model architectures from the library. For example models from sources such as BAAI, nomic, salesforce research, etc.
```
Comment (sankalp1999) [inline docs/src/embeddings/default_embedding_functions.md:151]: nitpick:
You can also load various other model architectures from the library, including those from BAAI, Nomic, Salesforce Research, and more.

File: docs/src/embeddings/default_embedding_functions.md (hunk 1 above)
Comment (AyushExel) [reply docs/src/embeddings/default_embedding_functions.md]: the messages get squished, only PR titles are visible when merged

Comment (sankalp1999) [reply docs/src/embeddings/default_embedding_functions.md]: yeah my bad lol

File: docs/src/embeddings/default_embedding_functions.md [hunk 5]
```diff
@@ -19,27 +19,163 @@ Allows you to set parameters when registering a `sentence-transformers` object.
...
+    - sentence-transformers/nq-distilbert-base-v1
+    - sentence-transformers/paraphrase-MiniLM-L12-v2
+    - sentence-transformers/paraphrase-MiniLM-L3-v2
+    - sentence-transformers/paraphrase-MiniLM-L6-v2
+    - sentence-transformers/paraphrase-TinyBERT-L6-v2
+    - sentence-transformers/paraphrase-albert-base-v2
+    - sentence-transformers/paraphrase-albert-small-v2
+    - sentence-transformers/paraphrase-distilroberta-base-v1
+    - sentence-transformers/paraphrase-distilroberta-base-v2
+    -transformers/paraphrase-multilingual-MiniLM-L12-v2
```
Comment (tanaymeh) [inline docs/src/embeddings/default_embedding_functions.md:130]: Should be `- transformers/paraphrase-multilingual-MiniLM-L12-v2`

**Comment by tanaymeh** [comment]
Looks good to me, @AyushExel

File: docs/src/embeddings/default_embedding_functions.md (hunk 5 above)
Comment (tanaymeh) [inline docs/src/embeddings/default_embedding_functions.md:130]:
```suggestion
    - transformers/paraphrase-multilingual-MiniLM-L12-v2
```

**Review by tanaymeh — APPROVED** [review]

**Pull request merged**